src/sentry/static/sentry/app/views/settings/organizationRelay/modals/modalManager.tsx

```diff
@@ -53,6 +54,17 @@ class DialogManager<
     }
   }
 
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.onKeyDown);
+  }
+
+  onKeyDown = (event: KeyboardEvent) => {
+    const {isFormValid} = this.state;
+    if (event.code === 'Enter' && isFormValid) {
+      this.handleSave();
```

Comment:
I think if you wrap the form in an actual form element, you will get this functionality natively, no?

Reply:
yes I can also use the onKeyDown of a form element...

Reply:
I meant more like `<Form onSubmit={this.handleSave}...`.
That way you don't need to mess with comparing event codes.

Reply:
@matejminar this would not work as the "Register" button is outside of the form

Reply:
There are ways around it (like form attribute on the button), but how are we doing it in other places? I know it works in debug files settings when adding a new repository.

Reply:
they have a button submit inside of the form... I tried to introduce the Form component in my Code, but I would need to change a lot of things..so I decided to don't do it right now. I've implemented your suggestion (form attribute on the button).